lib/queue.js

```diff
@@ -581,26 +577,30 @@ Queue.prototype.moveUnlockedJobsToWait = function(){
   console.log('moveUnlockedJobsToWait')
   var _this = this;
 
-  if(this.closing){
-    return this.closing;
-  } else{
-    return scripts.moveUnlockedJobsToWait(this).then(function(responses){
-      var handleFailedJobs = responses[0].map(function(jobId){
-        return _this.getJobFromId(jobId).then(function(job){
-          _this.distEmit('failed', job.toJSON(), new Error('job stalled more than allowable limit'));
-          return null;
-        });
+  return scripts.moveUnlockedJobsToWait(this).then(function(responses){
+    var handleFailedJobs = responses[0].map(function(jobId){
```

Comment:
I tried this, and it seems that spread must have an array, but in bull sometimes scripts.moveUnlockedJobsToWait returns undefined:
```js
 Cluster
Worker spawned: # 1
Worker spawned: # 2
Worker spawned: # 3
Worker spawned: # 4
Worker spawned: # 5
Worker spawned: # 6
Worker spawned: # 7
Failed to handle unlocked job in active: TypeError: Cannot read property 'map' of undefined
    at /home/charles/source/bull/lib/queue.js:576:40
    at tryCatcher (/home/charles/source/bull/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/charles/source/bull/node_modules/bluebird/js/release/promise.js:507:35)
    at Promise._settlePromise (/home/charles/source/bull/node_modules/bluebird/js/release/promise.js:567:18)
    at Promise._settlePromise0 (/home/charles/source/bull/node_modules/bluebird/js/release/promise.js:612:10)
    at Promise._settlePromises (/home/charles/source/bull/node_modules/bluebird/js/release/promise.js:691:18)
```